### PR TITLE
JSON key changes

### PR DESCRIPTION
**Pull request by cs-yash-patodkar**
Changes:
1. Keys have been changed for the JSON
2. Console log error added
3. "?" changed to "Key not found"

### CODE REVIEW

File: widget/view.controller.js [hunk 1]
```diff
@@ -472,7 +475,7 @@
               }
               $q.all(promises).then(function () {
                 var sortedDataSource = sortObjectByKeys(updatedObject);
-                sortedDataSource = Object.fromEntries(Object.entries(sortedDataSource).slice(0,3));
+                sortedDataSource = Object.fromEntries(Object.entries(sortedDataSource).slice(0, 3));
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:479]: Good practice: add comment why do you need slice(0,3) with an example?

File: widget/view.controller.js [hunk 2]
```diff
@@ -815,10 +818,10 @@
           if (result && result['hydra:member'] && result['hydra:member'].length > 0) {
             $scope.socResult.automatedClosed = result['hydra:member'][0].status;
           }
-          if($scope.socResult.closedAlerts === 0){
+          if ($scope.socResult.closedAlerts === 0) {
             var closedAutomatedAlerts = '(0%';
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js]: please always use const for hard coded strings like (0%

File: widget/view.controller.js [hunk 3]
```diff
@@ -815,10 +818,10 @@
...
           }
-          else{
+          else {
             var closedAutomatedAlerts = '( ' + Math.round(($scope.socResult.automatedClosed * 100) / $scope.socResult.closedAlerts) + '%';
           }
           addLabelCounts({ 'id': 'idResolvedAutomated', 'count': closedAutomatedAlerts, 'title': $scope.config.automatedResolved.title + ' )' });
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:829]: please const for repeated hard coded values like title, count, id, etc.

File: widget/assets/socWidgetInput.json [hunk 4]
```diff
@@ -86,165 +86,186 @@
   "impactAnalysis": [
     {
       "id": "idArtifactsDivision",
-      "title": "Artifacts analysed",
-      "value": "?"
+      "idFromUser": "artifactsAnalysed",
+      "title": "Key not found",
```
Comment (cs-suyog-jinturkar) [inline widget/assets/socWidgetInput.json]: 1. why title has changed to error value?
2. why do you need idFromUser field?
3. Please remove id from idArtifactsDivision as discussed
4. IMO, you can use const in js code for a common error message like "key not found", having it in json doesn't add any value

File: widget/view.controller.js (hunk 1 above)
Comment (cs-yash-patodkar) [reply widget/view.controller.js:479]: Done

File: widget/view.controller.js (hunk 2 above)
Comment (cs-yash-patodkar) [reply widget/view.controller.js]: Done

File: widget/assets/socWidgetInput.json (hunk 4 above)
Comment (cs-yash-patodkar) [reply widget/assets/socWidgetInput.json]: idFromUser to map, required iD's with id's provided by user, I have changed the id's as discussed

**Review by cs-suyog-jinturkar — APPROVED** [review]

**Pull request merged**